src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,903 @@
+//		     {
+//				 std::cout << "face " << face_no << " is at boundary" << std::endl;
+//		         typename dealii::DoFHandler<PHILIP_DIM>::active_cell_iterator neighbor = current_cell->neighbor_or_periodic_neighbor(face_no);
+//		         std::cout << "the neighbor is " << neighbor->index() << std::endl;
+//		     }
+//		 }
 //
-// }
-
- std::cout << "Implement initial conditions" << std::endl;
- dealii::FunctionParser<2> initial_condition;
- std::string variables = "x,y";
- std::map<std::string,double> constants;
- constants["pi"] = dealii::numbers::PI;
- std::string expression = "exp( -( 20*(x-1)*(x-1) + 20*(y-1)*(y-1) ) )";//"sin(pi*x)*sin(pi*y)";
- initial_condition.initialize(variables,
-         expression,
-         constants);
- dealii::VectorTools::interpolate(dg->dof_handler,initial_condition,dg->solution);
- // Create ODE solver using the factory and providing the DG object
- std::shared_ptr<PHiLiP::ODE::ODESolverBase<dim, double>> ode_solver = PHiLiP::ODE::ODESolverFactory<dim, double>::create_ODESolver(dg);
-
- double finalTime = 1.5;
-
- //double dt = all_parameters->ode_solver_param.initial_time_step;
- ode_solver->advance_solution_time(finalTime);
-
- return 0; //need to change
+//	}
+
+	std::cout << "Implement initial conditions" << std::endl;
+	
+//if use curvilinear grid to interpolate IC
+                const unsigned int n_quad_pts1      = dg->volume_quadrature_collection[poly_degree].size();
+                const unsigned int n_dofs_cell1     =dg->fe_collection[poly_degree].dofs_per_cell;
+         //   dealii::QGauss<dim> quad_val(poly_degree+1);
+            //dealii::QGaussLobatto<dim> quad_val(poly_degree+1);
+           // const dealii::Mapping<dim> &mapping = (*(dg->high_order_grid.mapping_fe_field));
+           // const dealii::MappingQGeneric<dim, dim> mapping(poly_degree+1);
+            dealii::FEValues<dim,dim> fe_values_test(*(dg->high_order_grid->mapping_fe_field), dg->fe_collection[poly_degree], dg->volume_quadrature_collection[poly_degree], 
+           // dealii::FEValues<dim,dim> fe_values_test(mapping, dg->fe_collection[poly_degree], dg->volume_quadrature_collection[poly_degree], 
+            //        dealii::update_values | dealii::update_JxW_values | dealii::update_quadrature_points);
+                                dealii::update_values | dealii::update_JxW_values | 
+                                dealii::update_jacobians |  
+                                dealii::update_quadrature_points | dealii::update_inverse_jacobians);
+           // const unsigned int n_quad_pts1 = fe_values_test.n_quadrature_points;
+            const unsigned int max_dofs_per_cell = dg->dof_handler.get_fe_collection().max_dofs_per_cell();
+            std::vector<dealii::types::global_dof_index> current_dofs_indices(max_dofs_per_cell);
+            const double pi2 = atan(1)*4.0;
+            for (auto current_cell = dg->dof_handler.begin_active(); current_cell!=dg->dof_handler.end(); ++current_cell) {
```

Comment:
This is just about moving this to another function to declutter advection_explicit_periodic. There is already a lot of code in the single function. The more we can decouple those, the better it is. Harder to clean up later on.

The issue is still valid, and ideally we would have a general function that can be used to initialize other functions as well.

Reply:
I agree entirely, but I think it will be quite involved for this pull request and would be a stand-alone pull request after. If that's alright? It will be the first issue I tackle after this pull request.

Reply:
It's quite literally copy pasting the those lines into a function somewhere else.

However if you think it will be more involved than this, then sure it can be done later on. It's not a high-priority issue, just cleaniness of code.

Reply:
Ok I decided to do this, and have pushed it with this PR.
Fixed.